lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -244,14 +254,15 @@ def validate_definitions(resources)
 
     def discover_resources
       resources = []
+      crds = cluster_resource_discoverer.crds(@sync_mediator)
       @logger.info("Discovering templates:")
 
       Dir.foreach(@template_dir) do |filename|
         next unless filename.end_with?(".yml.erb", ".yml", ".yaml", ".yaml.erb")
 
         split_templates(filename) do |r_def|
           r = KubernetesResource.build(namespace: @namespace, context: @context, logger: @logger,
-                                       definition: r_def, statsd_tags: @namespace_tags)
+                                       definition: r_def, statsd_tags: @namespace_tags, crds: crds)
```

Comment:
I think passing CRDs in favour of not having to duplicate the build logic makes sense since you would have to still know which resources in build to create a generic k8s resource for (non crs and not implemented) and then the rest used for crs or hardcoded crs. Passing a variable for a check seems a valid approach to me